### DISCUSSION TITLE
How to get a node from a right-click in the TreeView

### DISCUSSION THREAD

**Discussion by ApaulMyLittleAirport** [AvaloniaUI/Avalonia#9730] [Q&A]
I want to get the node information of the TreeView from right click, but these TreeView events(OnClick\OnTapped\OnPointerReleased) seem to be triggered by left click, can anyone help me?
Please.....

**Comment by maxkatz6** [comment]
In the event arguments there is a Source property which returns control on which event was raised.
From this control you can iterate tree up (using control.VisualParent or `control.FindAncestorOfType<TreeViewItem>()`)
